src/System.CommandLine/CommandLineConfiguration.cs

```diff
@@ -36,6 +36,17 @@ public class CommandLineConfiguration
 
             foreach (var symbol in symbols)
             {
+                foreach (var childSymbol in symbol.Children)
```

Comment:
I don't believe this is correct. The `symbols` [that are passed in](https://github.com/dotnet/command-line-api/blob/master/src/System.CommandLine/Builder/CommandLineBuilder.cs#L33) will likely just be the root command. The `Children` will only be its direct children. We need to iterate down through all of them to do the validation.

Reply:
@Keboo Just to clarify this, starting at `symbol.Children` is correct. We just need to make sure to recursively check the children below.

Reply:
Correct, just needs to be recursive.